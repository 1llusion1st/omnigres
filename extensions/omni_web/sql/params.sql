-- params type
select array ['key', 'value']::omni_web.params;
-- Invalid params should fail:
select array ['key']::omni_web.params;

\pset null '<null>'
select omni_web.param_get(omni_web.parse_query_string('a=1&a=2'), 'a');
select omni_web.param_get(omni_web.parse_query_string('a&a=2'), 'a');
select omni_web.param_get_all(omni_web.parse_query_string('a&a=2'), 'a');

-- Shortcuts
select omni_web.param_get('a=1', 'a');
select omni_web.param_get(convert_to('a=1', 'utf8'), 'a');
select omni_web.param_get_all('a=1&a=2', 'a');
select omni_web.param_get_all(convert_to('a=1&a=2', 'utf8'), 'a');
