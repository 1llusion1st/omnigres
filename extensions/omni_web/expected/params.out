-- params type
select array['key', 'value']::omni_web.params;
    array    
-------------
 {key,value}
(1 row)

-- Invalid params should fail:
select array['key']::omni_web.params;
ERROR:  value for domain omni_web.params violates check constraint "params_check"
\pset null '<null>'
select omni_web.param_get(omni_web.parse_query_string('a=1&a=2'), 'a');
 param_get 
-----------
 1
(1 row)

select omni_web.param_get(omni_web.parse_query_string('a&a=2'), 'a');
 param_get 
-----------
 <null>
(1 row)

select omni_web.param_get_all(omni_web.parse_query_string('a&a=2'), 'a');
 param_get_all 
---------------
 <null>
 2
(2 rows)

